### DISCUSSION TITLE
Should we necessarily use blank nodes?

### DISCUSSION THREAD

**Discussion by candlecao** [DDMAL/linkedmusic-datalake#120] [General]
First of all, I agree to use blank nodes wherever it's necessary.
This topic regards several points:

1. Some existing issues
> https://github.com/DDMAL/linkedmusic-datalake/issues/118<br>
>> at the bottome of the issue, we encounter this question

In addition, in the sets.csv file of The Session Database, I also encounter situation where blanks nodes are applicable:
<html xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office" xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40">
<head>

<meta name=Generator content="Microsoft Excel">

<style>

</style>
</head>
<body>

tuneset_id | name | tune_id | setting_id | type | type_wiki
-- | -- | -- | -- | -- | --
https://thesession.org/members/128154/sets/22927 | Rakes Of Mallow, The | https://thesession.org/tunes/85 | 12595 | polka | https://www.wikidata.org/wiki/Q153071
https://thesession.org/members/128154/sets/22927 | My Darling Asleep | https://thesession.org/tunes/76 | 76 | jig | https://www.wikidata.org/wiki/Q1079270
https://thesession.org/members/128154/sets/22927 | Drops Of Brandy, The | https://thesession.org/tunes/388 | 388 | jig | https://www.wikidata.org/wiki/Q1079270
https://thesession.org/members/128154/sets/22927 | Fanny Power | https://thesession.org/tunes/957 | 957 | waltz | https://www.wikidata.org/wiki/Q8701407
https://thesession.org/members/128154/sets/22927 | Brian Boru's March | https://thesession.org/tunes/271 | 13010 | jig | https://www.wikidata.org/wiki/Q1079270
https://thesession.org/members/128154/sets/22927 | Saint Patrick's Day | https://thesession.org/tunes/385 | 13216 | jig | https://www.wikidata.org/wiki/Q1079270
https://thesession.org/members/128154/sets/22927 | Planxty Irwin | https://thesession.org/tunes/790 | 13930 | waltz | https://www.wikidata.org/wiki/Q8701407
https://thesession.org/members/128154/sets/22927 | Irish Washerwoman, The | https://thesession.org/tunes/92 | 21778 | jig | https://www.wikidata.org/wiki/Q1079270
https://thesession.org/members/128154/sets/22927 | Tenpenny Bit, The | https://thesession.org/tunes/109 | 27514 | jig | https://www.wikidata.org/wiki/Q1079270
https://thesession.org/members/128154/sets/22927 | S铆 Bheag S铆 Mh贸r | https://thesession.org/tunes/449 | 28390 | waltz | https://www.wikidata.org/wiki/Q8701407
https://thesession.org/members/128154/sets/22927 | Tabhair Dom Do L谩mh | https://thesession.org/tunes/454 | 31867 | waltz | https://www.wikidata.org/wiki/Q8701407
https://thesession.org/members/128154/sets/22927 | Planxty Hewlett | https://thesession.org/tunes/1468 | 32625 | waltz | https://www.wikidata.org/wiki/Q8701407

</body>

</html>
@Yueqiao12Zhang We may talk about this.
> https://github.com/DDMAL/linkedmusic-datalake/issues/95<br>

**Comment by candlecao** [comment]
2. What if we use blank nodes, our python code for CSV2RDF probably have to be modified, I think. @Yueqiao12Zhang 
> Unless we modify the structure of CSVs for any given database, for example, using json in a cell for a attribute.

3. There may be “relation sets” in addition to “entity sets” in a relational database. As long as a relational database inevitably uses relation sets (and each relation set having its own unique attributes), it becomes necessary to use blank nodes in terms of converting the relational database into RDF database. Furthermore, a relation set may have a composite primary key composed of multiple fields, a scenario that our current code does not yet accommodate.

**Reply by fujinaga** [reply]
I prefer using named graphs. To be discussed further. Remember that I value provenance of each statement.

**Reply by candlecao** [reply]
> I prefer using named graphs. To be discussed further. Remember that I value provenance of each statement.

Hi Ich, I should take some time to look into how to use named graph to solve the issue practically.

**Reply by fujinaga** [reply]
Yes, please.
